docs/guides/blocks/formulas.md

```diff
@@ -30,20 +32,74 @@ Formulas can accept parameters, which is the primary way for them to access data
 
 ## Results
 
-TODO
+All formulas must return a result, which is a single value matching the type specified in the `resultType` property of the formula definition. See the [Data types guide][data-types] for more information on the type of values that can be returned.
+
+
+## Authentication
+
+Formulas can use [authentication][authentication] to fetch private data. When using [system authentication][system_auth] there is no change to the user experience, but when you use [user authentication][user_auth] the formula editor will be automatically updated to prompt for a connected account. The connected account will appear as the first parameter of the formula, but it's value is not passed to your formula's `execute` method.
+
+=== "Formula editor"
+    <img src="../../../images/formula_account.png" srcset="../../../images/formula_account_2x.png 2x" class="screenshot" alt="Account parameter in formula editor">
+
+=== "Code"
+    ```ts
+    pack.setUserAuthentication({
+      type: coda.AuthenticationType.HeaderBearerToken,
+      instructionsUrl: "https://todoist.com/app/settings/integrations",
+    });
+
+    pack.addFormula({
+      name: "GetTask",
+      description: "Gets a Todoist task by URL",
+      parameters: [
+        coda.makeParameter({
+          type: coda.ParameterType.String,
+          name: "url",
+          description: "The URL of the task",
+        }),
+      ],
+      // ...
+      execute: async function ([url], context) {
+        // ...
+      },
+    });
+    ```
+
+By default if a Pack defines an authentication method then all formulas will require a connected account. You can change this behavior using the [`connectionRequirement`][connectionRequirement] property of the formula definition. Setting it to `Optional` makes the user account optional, and `None` removes the account parameter completely.
+
+While it's possible to use multiple accounts within a document, each instance of the formula must use a specific account. It isn't possible to have a formula that uses the account of whichever user is currently viewing the document. It's important for collaboration that all users in a Coda doc have access to the same data.
 
 
 ## Caching
 
-TODO
+For performance reasons all formula results are cached by default. If your formula is called again with the same parameters, the result will be loaded from the cache instead of re-running your code. Building or releasing a new version of your Pack will invalidate this cache, ensuring you get fresh results using your new code.
```

Comment:
Again, only non-action formulas are cached by default.

Also, "the result will be loaded from the cache" makes it seem like it's guaranteed that subsequent invocations will get loaded from the cache, but they only _might_ be loaded from the cache, if it's still within the TTL and the cache hasn't grown too large.

Reply:
Ditto on actions.

Adjusted the wording to indicate that it attempts to load from the cache.